nf_core/modules/modules_json.py

```diff
@@ -432,9 +432,9 @@ def check_up_to_date(self):
         """
         self.load()
         if not self.has_git_url_and_base_path():
-            raise UserWarning(
-                "The 'modules.json' file is not up to date. " "You can fix it by running 'nf-core modules update'."
-            )
+            log.info("The 'modules.json' file is not up to date. Recreating the 'module.json' file.")
+            self.create()
+            self.load()
```

Comment:
I'd say it would be better with a try catch block around the call to `self.load` that catches the `FileNotFoundError` that it will throw if the file does not exist. You could also assign the `modules_json` variable to `self.modules_json` in `create` and skip the second call to `load` since there should be no need to go via the file.

Reply:
Good idea, changed it accordingly.
Now I am actually wondering, if we need all of the 
```
if self.modules_json is None:
    self.load()
```
in the different submethods, when they are (as far as I saw) all called after a `check_up_to_date()` call. Or is that just to be extra sure for future methods?

Reply:
Sounds good. Just grep for the creation of `ModulesJson` objects and make sure that they are not created without `check_up_to_date` being called. This will atleast be true for some tests.

Reply:
I think I will actually leave it for now. Too many possible downstream effects, I might not be aware of now. But I added a test for `.check_up_to_date()`, so this PR should be good to go. 🙂